crates/metaslang/cst/src/lib.rs

```diff
@@ -3,41 +3,46 @@ pub mod cursor;
 pub mod query;
 pub mod text_index;
 
-pub trait Kind:
-    Sized
-    + Copy
-    + Clone
-    + PartialEq
-    + Eq
-    + std::fmt::Display
-    + std::fmt::Debug
-    + serde::Serialize
-    + for<'a> std::convert::TryFrom<&'a str, Error = strum::ParseError>
-{
+pub trait AbstractKind: Sized + std::fmt::Debug + Copy + PartialEq + Eq + serde::Serialize {
+    fn as_static_str(&self) -> &'static str;
```

Comment:
Not sure I understand the reason behind these two new methods, over implementing the standard `Into<'static str>` and `TryFrom<&str>`.

Reply:
this is the solution to the strum problem and the fact that constraints on associated types are an unstable rust feature